src/grpc/charts.rs

```diff
@@ -51,7 +56,21 @@ impl chart_server::Chart for ChartService {
             }
 
             Ok(chart) => {
-                let ordered_chart_data = chart.data.into_iter().map(|cd| cd.into()).collect();
+                let ordered_chart_data: Vec<PbChartData> =
+                    try_join_all(chart.data.into_iter().map(|chart_data| async {
+                        let snap_name = get_snap_name(
+                            &chart_data.rating.snap_id,
+                            &self.ctx.config.snapcraft_io_uri,
+                            &self.ctx.http_client,
+                        )
+                        .await?;
+
+                        Result::<PbChartData, Error>::Ok(
+                            PbChartData::from_chart_data_and_snap_name(chart_data, &snap_name),
```

Comment:
rather than take a reference here for `snap_name` and then call `.into()` inside of `from_rating_and_snap_name` we should just pass the String all the way through.